sys/net/sock/sock_util.c

```diff
@@ -181,3 +181,26 @@ int sock_udp_str2ep(sock_udp_ep_t *ep_out, const char *str)
 #endif
     return -EINVAL;
 }
+
+bool sock_udp_ep_equal(const sock_udp_ep_t *a, const sock_udp_ep_t *b)
+{
+    assert(a && b);
+
+    /* compare family and port */
+    if ((a->family != b->family) || (a->port != b->port)) {
+        return 0;
+    }
+
+    /* compare addresses */
+    switch (a->family) {
+#ifdef SOCK_HAS_IPV6
+        case AF_INET6:
+            return (memcmp(a->addr.ipv6, b->addr.ipv6, 16) == 0);
+
+#endif
+        case AF_INET:
+            return (memcmp(a->addr.ipv4, b->addr.ipv4, 4) == 0);
+        default:
+            return true;
```

Comment:
this returns true if ```a->family``` is unknown.

Reply:
what should it return? My take was that on Family unknown we simply ignore the address field. Maybe not the smartest thing?!

Reply:
false, otherwise it returns true for different addresses of not compiled-in family.

best would be -ENOTSUP.

Reply:
could go for: `int sock_udp_ep_equal(...)`
@return 0 -> equal
@return -ENOTSUP if family unknown

but what for not equal?

Reply:
hm.

if this is a three-state (equal, nonequal, error), "equal" is probably best represented as zero, because otherwise people write ```if (sock_udp_ep_equal(a, b))``` and never catch the error. In that case, renaming to "sock_udp_ep_cmp()" (as strcmp()) would be most ideomatic C, and only >0 (or 1) is left for non-equal...

Why not simply "raise AddressFamilyNotSupportedException()"?

Reply:
just played with it, going back to an int return value with a potential negative return value is not very practical, considering the most common use case for this function is probably something like
```c
if (sock_udp_ep_equal(a, b)) {
....
```

-> I now changed the return value for the default path in the switch statement to `false` and added a half-sentence on this to the return value desc in the doc